src/deploymentTemplate.ts

```diff
@@ -0,0 +1,149 @@
+                        var parsed = JSON.parse(body);
+                        let items: vscode.QuickPickItem[] = [];
+                
+                        for (var i in parsed)
+                        {
+                            // list only directories and skip known directories that don't contain templates
+                            if (parsed[i].type == "dir" && !parsed[i].name.startsWith('.')) {
+                                items.push({ label: parsed[i].name, description: null });
+                            }
+                        }
+                        vscode.window.showQuickPick(items).then(selection => {
+                            // the user canceled the selection
+                            if (!selection) {
+                                return;
+                            }        
+                            
+                            __this.retrieveTemplate(selection.label);
+                        });
+                    });
+                } else {
+                    vscode.window.showErrorMessage("Error: " + response.statusCode + " " + response.statusMessage);
+                }
+
+            }).on('error', function(e) {
+                vscode.window.showErrorMessage("Error: " + e);
+            });
+        
+    }
+
+    public retrieveTemplate(templateName: string) {
+        var http = require('https');
+        let __this = this;
+        let repo: string = utilities.getCodeConfiguration<string>(null, Constants.Config_deploymentTemplatesGitHubRepo);
+
+            http.get({
+                host: Constants.GitHubRawContentHost,
+                path: '/' + repo + '/master/' + templateName + '/azuredeploy.json',
+                headers: { 'User-Agent': 'VSC Ansible Extension'}
+                }, function(response) {
+                    if (response.statusMessage == "OK") {
+                        var body = '';
+                        response.on('data', function(d) {
+                            body += d;
+                        });
+                        response.on('end', function() {
+                            try {
+                                var parsed = JSON.parse(body);
+
+                                __this.createPlaybookFromTemplate("https://" + Constants.GitHubRawContentHost + "/" + repo + "/master/" + templateName + "/azuredeploy.json",
+                                                                parsed);
+                            } catch (e) {
+                                vscode.window.showErrorMessage("Failed to parse 'azuredeploy.json'");
+                            }
+                        });
+                    } else if (response.statusCode == 404) {
+                        vscode.window.showErrorMessage("Template file 'azuredeploy.json' not found.");
+                    } else {
+                        vscode.window.showErrorMessage("Error: " + response.statusCode + " " + response.statusMessage);
```

Comment:
add sentence to show what's the statuscode for?

Reply:
status message is additional info, so for example, user will see:
"error: 500 internal server error"

Reply:
i mean where's the statuscode from, e.g 'error: failed to get xxx template content, error code 500,...'

Reply:
ok, i will add this info